src/main/groovy/life/qbic/datamodel/datasets/MaxQuantRunResult.groovy

```diff
@@ -0,0 +1,212 @@
+package life.qbic.datamodel.datasets
+
+import life.qbic.datamodel.datasets.datastructure.files.DataFile
+import life.qbic.datamodel.datasets.datastructure.files.maxquant.AllPeptides
+import life.qbic.datamodel.datasets.datastructure.files.maxquant.Evidence
+import life.qbic.datamodel.datasets.datastructure.files.maxquant.ExperimentalDesignTemplate
+import life.qbic.datamodel.datasets.datastructure.files.maxquant.Parameters
+import life.qbic.datamodel.datasets.datastructure.files.maxquant.Peptides
+import life.qbic.datamodel.datasets.datastructure.files.maxquant.ProteinGroups
+import life.qbic.datamodel.datasets.datastructure.files.maxquant.RunParameters
+import life.qbic.datamodel.datasets.datastructure.files.maxquant.Summary
+import life.qbic.datamodel.datasets.datastructure.files.general.SampleIds
+
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.Method
+
+/**
+ * Class that holds information about the output of a maxQuantRun
+ *
+ * @since 2.10.0
+ */
+final class MaxQuantRunResult {
+
+    // Fully qualified domain name of the maxQuant file structure package
+    private final static String FQDN_FILES = "life.qbic.datamodel.datasets.datastructure.files.maxquant"
+    // Fully qualified domain name of the general file structure package
+    private final static String GENERAL_FILES = "life.qbic.datamodel.datasets.datastructure.files.general"
+
+    private final static Set maxQuantFileTypes = [
+            FQDN_FILES + ".AllPeptides",
+            FQDN_FILES + ".Evidence",
+            FQDN_FILES + ".ExperimentalDesignTemplate",
+            FQDN_FILES + ".Parameters",
+            FQDN_FILES + ".Peptides",
+            FQDN_FILES + ".ProteinGroups",
+            FQDN_FILES + ".RunParameters",
+            FQDN_FILES + ".Summary",
+            GENERAL_FILES + ".SampleIds"
+    ]
+
+    private AllPeptides allPeptides
+
+    private Evidence evidence
+
+    private ExperimentalDesignTemplate experimentalDesignTemplate
+
+    private Parameters parameters
+
+    private Peptides peptides
+
+    private ProteinGroups proteinGroups
+
+    private RunParameters runParameters
+
+    private Summary summary
+
+    private SampleIds sampleIds
+
```

Comment:
is there a reason that these fields are mutable?

Reply:
The fields are similiar to the ones defined in the NFCorePipelineResult Fields but you're right these can be set to final. Nice Catch! 👀